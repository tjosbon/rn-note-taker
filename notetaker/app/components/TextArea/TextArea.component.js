import React, { Component } from 'react';
import { TextInput } from 'react-native';
import styles from './TextArea.component.style';

class TextArea extends Component {
  state = {
    text : 'hello from non-os specific file'
  }
  render() {
    const {...extraProps} = this.props;
    return (
      <TextInput
        {...extraProps}
        style={[styles.textArea, extraProps.style]}
        multiline = {true}
        onChangeText={this._onChangeText}
        value={this.state.text}
      />
    );
  }
  _onChangeText = ( text ) => {
    this.setState( { text } );
  }
}

export default TextArea;