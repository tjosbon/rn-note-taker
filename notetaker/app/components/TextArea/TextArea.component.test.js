import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from './TextArea.component';

describe('Text Area', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TextArea />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});