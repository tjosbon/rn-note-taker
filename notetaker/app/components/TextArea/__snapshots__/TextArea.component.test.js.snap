// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Area renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  multiline={true}
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 100,
        "width": 200,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
  value="hello from non-os specific file"
/>
`;
